test/createAzureResource.test.ts

```diff
@@ -58,14 +57,74 @@ suite('Create Azure Resources', async function (this: ISuiteCallbackContext): Pr
         ext.tree.dispose();
     });
 
-    test('createFunctionApp (Basic)', async () => {
+    test('Create windows function app (Basic) and deploy JavaScript project', async () => {
+        const functionName: string = 'HttpTrigger';
         resourceGroupsToDelete.push(resourceName1);
-        await runWithFuncSetting('advancedCreation', undefined, async () => {
-            ext.ui = new TestUserInput([resourceName1]);
-            await vscode.commands.executeCommand('azureFunctions.createFunctionApp');
-            const createdApp: WebSiteManagementModels.Site = await webSiteClient.webApps.get(resourceName1, resourceName1);
-            assert.ok(createdApp);
+        await clearProjectFile(projectPath);
+        await runWithFuncSetting('projectLanguage', ProjectLanguage.JavaScript, async () => {
+            await runWithFuncSetting('advancedCreation', undefined, async () => {
+                ext.ui = new TestUserInput([resourceName1]);
+                await vscode.commands.executeCommand('azureFunctions.createFunctionApp');
+                const createdApp: WebSiteManagementModels.Site = await webSiteClient.webApps.get(resourceName1, resourceName1);
+                assert.ok(createdApp);
+                await runWithFuncSetting('projectRuntime', ProjectRuntime.v2, async () => {
+                    const templateId: string = 'HttpTrigger-JavaScript';
+                    const authLevel: string = 'function';
+                    await vscode.commands.executeCommand('azureFunctions.createNewProject', projectPath, ProjectLanguage.JavaScript, ProjectRuntime.v2, false, templateId, functionName, { AuthLevel: authLevel });
+                    await validateProject(projectPath, getJavaScriptValidateOptions(true));
+                });
+            });
+        });
+
+        ext.ui = new TestUserInput([resourceName1, 'Deploy']);
+        await vscode.commands.executeCommand('azureFunctions.deploy');
+        await delay(5000);
+        const getFunction: WebSiteManagementModels.FunctionEnvelopeCollection = await webSiteClient.webApps.listFunctions(resourceName1, resourceName1);
+        assert.equal(getFunction[0].name, `${resourceName1}/${functionName}`);
+        // Initialize the contents of the clipboard as "Undefined" and then get function URL by command
+        clipboard.writeSync(typeof (undefined));
+        ext.ui = new TestUserInput([resourceName1, functionName]);
+        await vscode.commands.executeCommand('azureFunctions.copyFunctionUrl');
```

Comment:
Sorry to have back track (I know it must be frustrating) but I think that we shouldn't test `azureFunctions.copyFunctionUrl`.

In the code below, we are leveraging the vscode API so we are actually testing VS Code's functionality, not really the extension.

https://github.com/microsoft/vscode-azurefunctions/blob/15d06f939593cb58b217b6d6d3c3470e26493203/src/commands/copyFunctionUrl.ts#L16-L18

You should just access the `triggerUrl` property instead.  In the error message on L91, you should include the `triggerUrl` so we can see if it was properly generated in the error message in case of failure.

We don't want to use `clipboardy` because we have to install another package for one test case.  Also, there is no guarantee that `clipboardy` will always work and may cause false positives.

Reply:
Do you mean we shouldn't use "azureFunctions.copyFunctionUrl" command?
       If "YES", then whether we use the triggerUrl property to get the URL or not? How to use the triggerUrl? Can you give us some guidance?
       If "NO", then don't we need to verify 
"await vscode.commands.executeCommand('azureFunctions.copyFunctionUrl')"? Do we need to get the URL via "await vscode.env.clipboard.readText()" for next step?

Reply:
Yes, we shouldn't use `azureFunctions.copyFunctionUrl`.  Instead, you should get the treeItem and access its `triggerUrl` property.  You should be able to use `ext.tree.findTreeItem` by using the full id.  I believe that the id can be constructed with the function app that you retrieve.

Then we can uninstall `clipboardy`.